PYIC-8648: Update triage page links in the login to app flow

AmritSidhu opened the pull request:
## Proposed changes
### What changed

- Currently, the app download flow includes content suggesting that users can "prove their identity another way" if they choose not to use the app.
- For regular ID proving journeys, this is correct – users may be able to continue via an alternative path.
- For app-only routes (e.g. COI, 6MFC, international addresses, MFA reset), this content is misleading because no alternative route exists.

- The `iphone` and `android` contexts determine which QR code to display. A quick win for adding another context is to append the new context to the existing one and search for keywords instead of matching the context exactly.

### Why did it change

- This creates user frustration and failure, as some users leave the journey assuming an alternative path exists, only to find they cannot complete their task.

### Issue tracking
<!-- Jira ticket & other docs, like RFCs -->

- [PYIC-8648](https://govukverify.atlassian.net/browse/PYIC-8648)

## Checklists

- [ ] READMEs and documentation up-to-date
- [ ] Browser/ unit/ Selenium tests have been written/ updated
- [ ] No risk of exposure: PII, credentials, etc through logs/ code
- [ ] Ensure added/updated routes have CSRF protection if required


[PYIC-8648]: https://govukverify.atlassian.net/browse/PYIC-8648?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AmritSidhu committed:
PYIC-8648: Universal content updates for the download pages

AmritSidhu committed:
PYIC-8648: Add dynamic content to the download pages

Currently, the app download flow includes content suggesting that users can "prove their identity another way" if they choose not to use the app.

For regular ID proving journeys, this is correct – users may be able to continue via an alternative path.
For app-only routes (e.g. COI, 6MFC, international addresses, MFA reset), this content is misleading because no alternative route exists.

AmritSidhu committed:
PYIC-8648: add runtime string guard to validatePhoneType to prevent type confusion

blakeyp commented on src/app/shared/contextHelper.ts:
```diff
@@ -1,13 +1,11 @@
-      `Context cannot be parsed as a phone type: ${context}`,
-    );
-  }
+export function validatePhoneType(context?: string): void {
```
It might be cleaner/clearer if we just make this a function that returns a `PHONE_TYPES` given context?

i.e. if context.startsWith("android") -> `PHONE_TYPES.ANDROID`

then `getAppStoreRedirectUrl` can still use an actual `PHONE_TYPES` rather than a context string pretending to be one

AmritSidhu committed:
PYIC-8648: change validatePhoneType to return PhoneType

This avoids using context string pretending to be phone type
This also reduces checks in getAppStoreRedirectUrl

blakeyp commented on src/constants/device-constants.ts:
```diff
@@ -1,7 +1,9 @@
-});
+} as const);
+
+export type PhoneType = (typeof PHONE_TYPES)[keyof typeof PHONE_TYPES];
```
We should be able to use a Typescript string enum for this (which can also be used as a type) ?
https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

Though we'd probably want to apply the same to all constants so happy for this to be a separate tech debt ticket

AmritSidhu committed:
PYIC-8648: replace PHONE_TYPES and OS_TYPES objects with TypeScript string enums

Switched PHONE_TYPES and OS_TYPES from frozen object literals to TypeScript string enums for cleaner syntax and stronger type safety.

blakeyp commented on src/constants/device-constants.ts:
```diff
@@ -1,13 +1,11 @@
-  IOS: "iOS",
-  ANDROID: "Android",
-});
+export enum OS_TYPES {
```
Minor but the convention for enums/type is usually singular noun `PHONE_TYPE` rather than plural

AmritSidhu committed:
PYIC-8648: fix enums/type to use singular noun

AmritSidhu committed:
PYIC-8648: Fix details ordering

AmritSidhu committed:
PYIC-8648: Fix bugs

- Use correct context in links
- Link to welsh version of pages
- Open link in new tab

AmritSidhu committed:
PYIC-8648: fix links and translations

AmritSidhu committed:
PYIC-8648: Update page links contexts
